lib/msal-common/src/client/BaseClient.ts

```diff
@@ -178,6 +171,51 @@ export abstract class BaseClient {
         return response;
     }
 
+    /**
+     * Wraps sendPostRequestAsync with necessary preflight and postflight logic
+     * @param thumbprint - Request thumbprint for throttling
+     * @param tokenEndpoint - Endpoint to make the POST to
+     * @param options - Body and Headers to include on the POST request
+     * @param correlationId - CorrelationId for telemetry
+     */
+    async sendPostRequest<T extends ServerAuthorizationTokenResponse>(
+        thumbprint: RequestThumbprint,
+        tokenEndpoint: string,
+        options: NetworkRequestOptions,
+        correlationId: string
+    ): Promise<NetworkResponse<T>> {
+        ThrottlingUtils.preProcess(this.cacheManager, thumbprint);
+
+        let response;
+        try {
+            response = await this.networkClient.sendPostRequestAsync<T>(
+                tokenEndpoint,
+                options
+            );
+            const responseHeaders = response.headers || {};
+            this.performanceClient?.addFields(
+                {
+                    refreshTokenSize: response.body.refresh_token?.length || 0,
+                    httpVerToken:
+                        responseHeaders[HeaderNames.X_MS_HTTP_VERSION] || "",
+                    requestId:
+                        responseHeaders[HeaderNames.X_MS_REQUEST_ID] || "",
+                },
+                correlationId
+            );
+        } catch (e) {
+            if (e instanceof AuthError) {
+                throw e;
+            } else {
+                throw createClientAuthError(ClientAuthErrorCodes.networkError);
+            }
+        }
+
+        ThrottlingUtils.postProcess(this.cacheManager, thumbprint, response);
```

Comment:
I know you just copy/pasted this from the old NetworkManager file, but I just want to confirm that this will be reached even if an error is thrown directly above?

Reply:
It will not, but that is expected. The throttling logic requires a network response which won't be available if the fetch call threw.